src/code/z_vr_box.c

```diff
@@ -32,12 +97,383 @@ typedef struct {
+            temp = D_8012ACA0[sp424][pad420];
+
+            roomVtx[arg2 + pad420].v.ob[0] = sp358[temp];
+            roomVtx[arg2 + pad420].v.ob[1] = sp2A4[temp];
+            roomVtx[arg2 + pad420].v.ob[2] = sp1F0[temp];
+            roomVtx[arg2 + pad420].v.flag = 0;
+            roomVtx[arg2 + pad420].v.tc[0] = sp13C[temp];
+            roomVtx[arg2 + pad420].v.tc[1] = sp88[temp];
+            roomVtx[arg2 + pad420].v.cn[1] = 0;
+            roomVtx[arg2 + pad420].v.cn[2] = 0;
+            roomVtx[arg2 + pad420].v.cn[0] = 255;
+        }
+        gSPVertex(skyboxCtx->unk_138++, &roomVtx[arg2], 32, 0);
+        arg2 += pad420;
+        gSPCullDisplayList(skyboxCtx->unk_138++, 0, 15);
+
+        for (phi_t2_4 = 0, phi_ra = 0; phi_ra < 4; phi_ra++, phi_a2_4 += 0x1F) {
+            for (phi_a0_4 = 0, phi_t1 = 0; phi_t1 < 4; phi_t1++, phi_a0_4 += 0x3F, phi_t2_4 += 4) {
+                gDPLoadTextureTile(skyboxCtx->unk_138++, (u32)skyboxCtx->staticSegments[0] + D_8012AC90[arg8],
+                                   G_IM_FMT_CI, G_IM_SIZ_8b, 256, 0, phi_a0_4, phi_a2_4, phi_a0_4 + 0x3F,
+                                   phi_a2_4 + 0x1F, 0, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOLOD,
+                                   G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOLOD);
+                gSP1Quadrangle(skyboxCtx->unk_138++, D_8012AD40[phi_t2_4 + 1], D_8012AD40[phi_t2_4 + 2],
+                               D_8012AD40[phi_t2_4 + 3], D_8012AD40[phi_t2_4 + 0], 3);
+            }
+        }
+        gSPEndDisplayList(skyboxCtx->unk_138++);
+    }
+    return arg2;
+}
+#else
 s32 func_800ADBB0(SkyboxContext* skyboxCtx, Vtx* roomVtx, s32, UNK_TYPE, UNK_TYPE, UNK_TYPE, UNK_TYPE, UNK_TYPE, s32,
                   s32);
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_vr_box/func_800ADBB0.s")
+#endif
 
+#ifdef NON_MATCHING
+// Loops have very strange structure. In principle, they're the double loop over a 2D array shown below them.
+// However, that gives wildly different codegen and the part after the switch suggests a linear array.
+s32 func_800AE2C0(SkyboxContext* skyboxCtx, Vtx* roomVtx, s32 arg2, s32 arg3, s32 arg4, s32 arg5, s32 arg6, s32 arg7,
+                  s32 arg8) {
+    s32 pad334;
+    s32 pad330;
+    s32 i;
+    // 32C
+    s32 j;
+    // 328
+    s16 phi_a2_4;
+    s16 phi_a0_4;
+    // 324
+    s16 phi_t1;
+    u16 temp;
+    // 320
+    s16 sp320;
+    s16 sp31E;
+    // 31C
+    s16 phi_t2_4;
+    s16 phi_ra;
+    // 318
+    s32 sp2B4[5 * 5];
+    s32 sp250[5 * 5];
+    s32 sp1EC[5 * 5];
+    s32 sp188[5 * 5];
+    s32 sp124[5 * 5];
+    s32 *temp358;
+    s32 *temp2A4;
+    s32 *temp1F0;
+    s32 *temp13C;
+    s32 *temp88;
+    s16 temp3;
+    s32 pad428;
+    s32 pad42C;
+
+    switch (arg8) {
+        case 0:
+        case 1:
+            temp358 = sp2B4;
+            temp2A4 = sp250;
+            temp1F0 = sp1EC;
+            temp13C = sp188;
+            temp88 = sp124;
```

Comment:
this is omega confusing because the `temp*` and `sp*` variables are labeled differently
eg `temp358 = sp2B4;` should probably be `temp2B4 = sp2B4;`